pkg/cloudprovider/aws/node.go

```diff
@@ -26,12 +26,12 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type NodeFactory struct {
+type NodeAPI struct {
```

Comment:
I'm not sold on this naming change. This isn't an API, it's a https://en.wikipedia.org/wiki/Factory_method_pattern. Any time you see .For() in my code, it's a factory. Factories are hugely useful when necessary. In this case, the NodeFactory encapsulates some dependencies (e.g. EC2) vends node objects.

Reply:
I agree. Good point. I'll change this back.